File: backend/admin/router/auth_router/sys_dict.go
package auth_router

import (
	"admin/fiberc/handler"
	"admin/fiberc/middleware"
	"admin/router/logic"

	"github.com/gofiber/fiber/v3"
)

func registerSysDictRouters(router fiber.Router) {
	sysDictHandler := logic.SysDictHandler{}

	dictType := router.Group("/type")
	dictType.Post("/list", handler.CtxHandlerFunc(sysDictHandler.TypeList))
	logMiddleware := middleware.OperationLogMiddleware(middleware.WithModule("dict"))
	dictType.Post("/create", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.TypeCreate))
	dictType.Post("/update", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.TypeUpdate))
	dictType.Post("/switch", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.TypeSwitch))
	dictType.Post("/del", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.TypeDel))

	dictEntry := router.Group("/entry")
	dictEntry.Post("/list", handler.CtxHandlerFunc(sysDictHandler.EntryList))
	dictEntry.Post("/create", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.EntryCreate))
	dictEntry.Post("/update", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.EntryUpdate))
	dictEntry.Post("/switch", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.EntrySwitch))
	dictEntry.Post("/del", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.EntryDel))
	dictEntry.Post("/batch/copy", logMiddleware, handler.CtxHandlerNilFunc(sysDictHandler.EntryBatchCopy))
}

File: front/apps/admin-react/src/api/sysDict.ts
import API from './index'

export interface DictType {
  id: number
  typeCode: string
  typeName: string
  isEnabled: boolean
  sortOrder: number
  remark: string
  canWrite?: boolean
  canDelete?: boolean
}

export interface DictEntry {
  id: number
  labelComponent: string
  entryLabel: string
  entryValue: string
  languageCode: string
  sysDictTypeId: number
  sysDictType?: {
    typeCode: string
    typeName: string
  }
  sortOrder: number
  isEnabled: boolean
  remark: string
}

export interface DictMatchedEntry {
  id: number
  labelComponent: string
  entryLabel: string
  entryValue: string
}

export interface ReqDictEntryMatch {
  code: string
}

export interface ReqDictTypeCreate {
  typeCode: string
  typeName: string
  isEnabled: boolean
  sortOrder: number
  remark: string
}

export interface ReqDictTypeUpdate extends Partial<ReqDictTypeCreate> {
  id: number
}

export interface ReqDictTypeBatchDelete {
  ids: number[]
}

export interface ReqDictEntryCreate {
  labelComponent: string
  entryLabel: string
  entryValue: string
  languageCode: string
  sysDictTypeId: number
  sortOrder: number
  isEnabled: boolean
  remark: string
}

export interface ReqDictEntryUpdate extends Partial<ReqDictEntryCreate> {
  id: number
}

export interface ReqDictEntryBatchDelete {
  ids: number[]
}

export interface ReqDictEntryBatchCopy {
  entryIds: number[]
  targetTypeId: number
}

function typeList(req: PagingRequest) {
  return API.Post<Res<PagingResult<DictType>>>('/api/sys/dict/type/list', req, {
    cacheFor: 0,
  })
}

async function typeCreate(req: ReqDictTypeCreate) {
  await API.Post<Res>('/api/sys/dict/type/create', req, {
    cacheFor: 0,
  }).send()
}

async function typeUpdate(req: ReqDictTypeUpdate) {
  await API.Post<Res>('/api/sys/dict/type/update', req, {
    cacheFor: 0,
  }).send()
}

async function typeDel(req: ReqDictTypeBatchDelete) {
  await API.Post<Res>('/api/sys/dict/type/del', req, {
    cacheFor: 0,
  }).send()
}

function entryList(req: PagingRequest) {
  return API.Post<Res<PagingResult<DictEntry>>>('/api/sys/dict/entry/list', req, {
    cacheFor: 0,
  })
}

function entryMatch(req: ReqDictEntryMatch) {
  return API.Post<Res<DictMatchedEntry[]>>('/api/sys/dict/entry/match', req, {
    cacheFor: 0,
  })
}

async function entryCreate(req: ReqDictEntryCreate) {
  await API.Post<Res>('/api/sys/dict/entry/create', req, {
    cacheFor: 0,
  }).send()
}

async function entryUpdate(req: ReqDictEntryUpdate) {
  await API.Post<Res>('/api/sys/dict/entry/update', req, {
    cacheFor: 0,
  }).send()
}

async function entryDel(req: ReqDictEntryBatchDelete) {
  await API.Post<Res>('/api/sys/dict/entry/del', req, {
    cacheFor: 0,
  }).send()
}

async function entryBatchCopy(req: ReqDictEntryBatchCopy) {
  await API.Post<Res>('/api/sys/dict/entry/batch/copy', req, {
    cacheFor: 0,
  }).send()
}

export const DictApi = {
  typeList,
  typeCreate,
  typeUpdate,
  typeDel,
  entryList,
  entryMatch,
  entryCreate,
  entryUpdate,
  entryDel,
  entryBatchCopy,
}
